src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageDirectiveFeature.cs

```diff
@@ -9,20 +9,41 @@ namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
 {
     public static class PageDirectiveFeature
     {
-        public static bool TryGetRouteTemplate(RazorProjectItem projectItem, out string template)
+        public static bool TryGetPageDirective(RazorProjectItem projectItem, out string template)
         {
+            if (projectItem == null)
+            {
+                throw new ArgumentNullException(nameof(projectItem));
+            }
+
             const string PageDirective = "@page";
 
-            string content;
-            using (var streamReader = new StreamReader(projectItem.Read()))
+            var stream = projectItem.Read();
+
+            string content = null;
+            using (var streamReader = new StreamReader(stream))
             {
-                content = streamReader.ReadToEnd();
+                do
+                {
+                    content = streamReader.ReadLine();
+                } while (content != null && string.IsNullOrWhiteSpace(content));
+                content = content != null ? content.Trim() : content;
             }
 
-            if (content.StartsWith(PageDirective, StringComparison.Ordinal))
+            if (content != null && content.StartsWith(PageDirective, StringComparison.Ordinal))
             {
-                var newLineIndex = content.IndexOf(Environment.NewLine, PageDirective.Length);
-                template = content.Substring(PageDirective.Length, newLineIndex - PageDirective.Length).Trim();
+                template = content.Substring(PageDirective.Length, content.Length - PageDirective.Length).Trim();
```

Comment:
This second trim will execute after the Substring which removes "@page" and will trim off any spaces between `@page` and the first `"`. Should I make it `TrimStart` to make that more explicit?